src/backend/distributed/executor/multi_task_tracker_executor.c

```diff
@@ -452,7 +453,7 @@ MultiTaskTrackerExecute(Job *job)
  * execution to a list. The function then returns the list.
  */
 List *
-TaskAndExecutionList(List *jobTaskList)
+TaskAndExecutionList(List *jobTaskList, bool viaTaskTracker)
```

Comment:
This function name and comment is quite confusing now. If you pass `viaTaskTracker` you don't actually do anything with a `TaskExecution`. I'd suggest giving it a different name, like:
```c
CreateTaskList(List *jobTaskList, bool withTaskExecution)
```

And update the comment to include something about the `dependendTaskList`, because that's all it does for the adaptive executor afaict.